File: js/examples/python.mjs
import { readFile } from "node:fs/promises";

import { Wasmer } from "@wasmer/sdk2/node";

const source = await readFile(
  new URL("../../fixtures/python/hello.py", import.meta.url),
);

const client = new Wasmer();
const sandbox = await client.sandboxes.create({
  packages: ["python/python@3.13.5"],
  files: { "hello.py": source },
});

try {
  const output = await sandbox
    .command("python", ["/workspace/hello.py"])
    .run({ check: true });

  console.log(output.text().trim());
} finally {
  await sandbox.close();
  await client.close();
}

File: python/examples/python.py
import asyncio
from pathlib import Path

from wasmer_sdk import Wasmer


SOURCE = (
    Path(__file__).resolve().parents[2]
    / "fixtures/python/hello.py"
)


async def main() -> None:
    wasmer = Wasmer()
    sandbox = await wasmer.sandboxes.create(
        packages=["python/python@=3.13.18"],
        files={"hello.py": SOURCE.read_bytes()},
    )
    output = await sandbox.command(
        "python", ["/workspace/hello.py"]
    ).run()
    print(output.text().strip())


asyncio.run(main())
